File: frontend/src/services/authService.ts
import api from './api';
import { ApiResponse, LoginResponse, Usuario } from '../types';

export async function login(email: string, senha: string) {
  const { data } = await api.post<ApiResponse<LoginResponse>>('/auth/login', {
    email, senha,
  });
  return data.data;
}

export async function logout() {
  try { await api.post('/auth/logout'); } catch { /* ignore */ }
}

export async function me() {
  const { data } = await api.get<ApiResponse<Usuario>>('/usuarios/me');
  return data.data;
}

File: mobile/lib/data/services/auth_service.dart
import '../models/login_response.dart';
import '../models/usuario.dart';
import '../../core/network/api_client.dart';

class AuthService {
  static final _dio = ApiClient().dio;

  static Future<LoginResponse> login(String email, String senha) async {
    final resp = await _dio.post('/auth/login', data: {
      'email': email,
      'senha': senha,
    });
    return LoginResponse.fromJson(
      Map<String, dynamic>.from(resp.data['data'] as Map),
    );
  }

  static Future<void> logout() async {
    try {
      await _dio.post('/auth/logout');
    } catch (_) {/* ignorar */}
  }

  static Future<Usuario> me() async {
    final resp = await _dio.get('/usuarios/me');
    return Usuario.fromJson(
      Map<String, dynamic>.from(resp.data['data'] as Map),
    );
  }
}
